File: backend/src/main/java/com/school/saas/module/student/dto/StudentDetailDTO.java
package com.school.saas.module.student.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class StudentDetailDTO {
    private UUID id;
    private UUID userId;
    private String firstName;
    private String lastName;
    private String email;
    private String phoneNumber;
    private UUID schoolId;
    private String schoolName;
    private UUID classRoomId;
    private String classRoomName;
    private String classRoomLevel;
    private String classRoomSection;
    private String registrationNumber;
    private LocalDate birthDate;
    private String gender;
    private LocalDate enrollmentDate;
    private String status;
    private String address;
    private String avatarUrl;
    private String administrativeDocuments;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: mobile/lib/models/student_detail_dto.dart
class StudentDetailDTO {
  final String id;
  final String userId;
  final String firstName;
  final String lastName;
  final String email;
  final String schoolId;
  final String classRoomId;
  final String classRoomName;
  final String classRoomLevel;
  final String classRoomSection;
  final String registrationNumber;
  final String birthDate;
  final String gender;
  final String enrollmentDate;
  final String status;

  StudentDetailDTO({
    required this.id,
    required this.userId,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.schoolId,
    required this.classRoomId,
    required this.classRoomName,
    required this.classRoomLevel,
    required this.classRoomSection,
    required this.registrationNumber,
    required this.birthDate,
    required this.gender,
    required this.enrollmentDate,
    required this.status,
  });

  factory StudentDetailDTO.fromJson(Map<String, dynamic> json) {
    return StudentDetailDTO(
      id: json['id'],
      userId: json['userId'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      schoolId: json['schoolId'],
      classRoomId: json['classRoomId'],
      classRoomName: json['classRoomName'],
      classRoomLevel: json['classRoomLevel'],
      classRoomSection: json['classRoomSection'],
      registrationNumber: json['registrationNumber'],
      birthDate: json['birthDate'],
      gender: json['gender'],
      enrollmentDate: json['enrollmentDate'],
      status: json['status'],
    );
  }
}
